packages/flutter/test/cupertino/text_field_test.dart

```diff
@@ -51,6 +72,30 @@ void main() {
     },
   );
 
+  testWidgets(
+    'multi-lined text fields are intrinsically taller no-strut',
+    (WidgetTester tester) async {
+      await tester.pumpWidget(
+        CupertinoApp(
+          home: Center(
+            child: ConstrainedBox(
+              constraints: BoxConstraints.loose(const Size(200, 200)),
+              child: const CupertinoTextField(
+                maxLines: 3,
+                strutStyle: StrutStyle.disabled(),
+              ),
+            ),
+          ),
+        ),
+      );
+
+      expect(
+        tester.getSize(find.byType(CupertinoTextField)),
+        const Size(200, 63),
```

Comment:
Appreciated the last comment about what 29 means. What does 63 mean? It's not 23*3...

Similar comments apply to the size checks in the following tests

Reply:
I gave a breakdown of what values are added to what for the ones where it was applicable.